lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+          end
+        end
+
+        def order_string
+          if @scope.order_values.any?
+            orders = @scope.order_values.join(', ')
+            "ORDER BY #{orders}"
+          elsif @scope.id_values.many?
+            id_orders = @scope.id_values.map { |id| "ID=#{@adapter.quote(id)} DESC" }.join(',')
+            "ORDER BY #{id_orders}"
+          end
+        end
+
+        def limit_string
+          if @scope.limit_value
+            "LIMIT #{@scope.limit_value}"
+          end
+        end
+      end
+
+      def primary_key_column
+        'id'
+      end
+
+      def connection
+        # conf = {:adapter=>"postgresql", :encoding=>"unicode", :database=>"axle_place_test", :pool=>5, :username=>"postgres"}
+        # @connection ||= ActiveRecord::Base.postgresql_connection(conf)
+        ActiveRecord::Base.connection
+      end
+
+      def execute(statement)
+        ActiveSupport::Notifications.instrument("cql.cassandra_object", cql: statement) do
+          connection.execute statement
+        end
+      end
+
+      def select(scope)
+        statement = QueryBuilder.new(self, scope).to_query
+
+        connection.execute(statement).each do |attributes|
+          yield attributes[primary_key_column], Oj.compat_load(attributes['attribute_store'])
+        end
+      end
+
+      def insert(table, id, attributes)
+        not_nil_attributes = attributes.reject { |key, value| value.nil? }
+        statement = "INSERT INTO #{table} (#{primary_key_column}, attribute_store) VALUES (#{quote(id)}, #{to_quoted_jsonb(not_nil_attributes)})"
+        execute_batchable statement
+      end
+
+      def update(table, id, attributes)
+        return if attributes.empty?
+
+        not_nil_attributes = attributes.reject { |key, value| value.nil? }
+        nil_attributes = attributes.select { |key, value| value.nil? }
+
+        if not_nil_attributes.any? && nil_attributes.any?
+          value_update = "jsonb_merge(jsonb_delete(attribute_store, #{fields_to_postgres_array(nil_attributes.keys)}), #{to_quoted_jsonb(not_nil_attributes)})"
+        elsif not_nil_attributes.any?
+          value_update = "jsonb_merge(attribute_store, #{to_quoted_jsonb(not_nil_attributes)})"
+        elsif nil_attributes.any?
+          value_update = "jsonb_delete(attribute_store, #{fields_to_postgres_array(nil_attributes.keys)})"
+        end
+
+        statement = "UPDATE #{table} SET attribute_store = #{value_update} WHERE #{primary_key_column} = #{quote(id)}"
+        execute_batchable statement
+      end
+
+      def delete(table, ids)
+        statement = "DELETE FROM #{table} WHERE #{create_ids_where_clause(ids)}"
+
+        execute_batchable statement
+      end
+
+      def execute_batch(statements)
+        stmt = [
+          "BEGIN",
+          statements * ";\n",
+          'COMMIT'
+        ] * ";\n"
+
+        execute stmt
+      end
+
+      def create_table(table_name, options = {})
+        define_jsonb_functions!
```

Comment:
The app will require some kind of migration. I started this by copying and pasting the hstore_adapter - just wanted to get something working.